Framework/DataHandling/test/LoadAscii2Test.h

```diff
@@ -542,4 +542,90 @@ class LoadAscii2Test : public CxxTest::TestSuite {
   size_t m_testno;
 };
 
+class LoadAscii2TestPerformance : public CxxTest::TestSuite {
+public:
+  void setUp() override {
+    setupFile();
+    loadAlg.initialize();
+
+    TS_ASSERT_THROWS_NOTHING(loadAlg.setPropertyValue("Filename", filename));
+    loadAlg.setPropertyValue("OutputWorkspace", outputName);
+    loadAlg.setPropertyValue("Separator", sep);
+    loadAlg.setPropertyValue("CustomSeparator", custsep);
+    loadAlg.setPropertyValue("CommentIndicator", comment);
+
+    loadAlg.setRethrows(true);
+  }
+
+  void testLoadAscii2Performance() {
+    TS_ASSERT_THROWS_NOTHING(loadAlg.execute());
+  }
+
+  void tearDown() override {
+    TS_ASSERT_THROWS_NOTHING(Poco::File(filename).remove());
+    AnalysisDataService::Instance().remove(outputName);
+  }
+
+private:
+  LoadAscii2 loadAlg;
+
+  const std::string outputName = "outWs";
+  std::string filename;
+
+  // Common saving/loading parameters
+  const std::string sep = "CSV";
+  const std::string custsep = "";
+  const std::string comment = "#";
+
+  void setupFile() {
+    constexpr int numVecs = 100;
+    constexpr int xyLen = 100;
+
+    Mantid::DataObjects::Workspace2D_sptr wsToSave =
+        boost::dynamic_pointer_cast<Mantid::DataObjects::Workspace2D>(
+            WorkspaceFactory::Instance().create("Workspace2D", numVecs, xyLen,
+                                                xyLen));
+
+    for (int i = 0; i < numVecs; i++) {
+      auto &X = wsToSave->mutableX(i);
+      auto &Y = wsToSave->mutableY(i);
+      auto &E = wsToSave->mutableE(i);
+      for (int j = 0; j < xyLen; j++) {
+        X[j] = 1.5 * j / 0.9;
+        Y[j] = (i + 1) * (2. + 4. * X[j]);
+        E[j] = 1.;
+      }
+    }
```

Comment:
No need to init this with fancy values for a performance test, all default should be fine in this case, I think.